README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
+
+* When the process is completed or exit out due to errors the log folder will have a timestamp appended so pids can be reused
+
+* A second response will be sent to indicate thats its finished with the outputs and error outputs
+
+* Logs will be outputted to the output and errOutput (up to a certain size from the top) of the response and sent to the client via a response when done 
+
+```
+type LogOutput {
+    //contents of log, (may change to string array)
+    string contents 
+}
+
+type StartRequest {
+    //path of a program or executable
+    string command
+    //arguments to invoke program or executable with
+    []string args 
+    // environment variables for the execution
+    //OPTIONAL variable
+    []string env
+    // current working directory of the execution
+    //OPTIONAL variable, default will be the working directory of server 
+    string dir 
+}
+
+type StartResponse {
+    //process Id of the running process executed by the command
+    //will be 0 if process failed to execute
+    int pid 
+    //stdout(standard output) of command/program executed
+    LogOutput output
+    //stderr(standard Error) of command/program executed
+    LogOutput ErrorOutput 
+    //if the command was finished it will be 0, otherwise if command successfully completes it will be 1
+    bool finished
+    //timestamp returns the starting time if just executed or finish time if finished
+    string timeStamp 
+}
+
+```
+#### Stop
+
+* The User should be able to stop the request based on the pid
+
+* When stopped the pid should be killed 
+
+* A response will be sent to client indicating process have been stopped along with the contents of the log
+
+```
+type StopRequest {
+    int pid
+}
+
+type StopResponse {
+    LogOutput Output
+    LogOutput errorOutput
+    bool isKilled
+}
+
+```
+
+#### Query 
+
+* The client should be able to check what processes are still running
+
+* The Client should be able fetch the logs and fetch the status of the process by pid and starting timestamp
+
+* The client should able to view list of all logs generated by the server
+
+* Client should be able to read the logs into memory and send it and will only grab up to a certain amout of bytes from
+the file to prevent maxing out string size or memory 
+
+
+* There will be only two types of Query
+
+    * QueryOneProcess:
+
+        * get information on one pid
+
+    * QueryRunningProcesses:
+
+        * get list of running processes from server
+
+    * QueryListOfLogs
+
+        * get list of all logs generated by server 
```

Comment:
this is not needed.
client should be able to list all processes and get the logs for any individual process.
getting logs for all processes at once won't be practical

Reply:
noted and will take out QueryListOfLogs